aerugo-hal/src/system_hal.rs

```diff
@@ -0,0 +1,57 @@
+//! System HAL.
+
+mod config;
+
+pub use self::config::SystemHardwareConfig;
+
+use bare_metal::CriticalSection;
+use core::ops::{Add, Sub};
+
+/// System HAL trait.
+pub trait SystemHal {
+    /// Type for an instant in time.
+    type Instant: Ord
+        + Copy
+        + Add<Self::Duration, Output = Self::Instant>
+        + Sub<Self::Duration, Output = Self::Instant>
+        + Sub<Self::Instant, Output = Self::Duration>;
```

Comment:
Why no `Add<Self::Instant, Output = Self::Duration>`? Is it automatically generated somehow?

Reply:
Adding `Duration` to `Instant` you still get an `Instant`.
![image](https://github.com/n7space/aerugo/assets/5027115/ca65ecb6-0b69-44d8-978b-e814d3e5d14f)